examples/animation/animated_transform.rs

```diff
@@ -49,79 +49,79 @@ fn setup(
 
     // Creating the animation
     let mut animation = AnimationClip::default();
-    // A curve can modify a single part of a transform, here the translation
+    // A curve can modify a single part of a transform: here, the translation.
     let planet_animation_target_id = AnimationTargetId::from_name(&planet);
     animation.add_curve_to_target(
         planet_animation_target_id,
-        VariableCurve::linear::<TranslationKeyframes>(
-            [0.0, 1.0, 2.0, 3.0, 4.0],
-            [
-                Vec3::new(1.0, 0.0, 1.0),
-                Vec3::new(-1.0, 0.0, 1.0),
-                Vec3::new(-1.0, 0.0, -1.0),
-                Vec3::new(1.0, 0.0, -1.0),
-                // in case seamless looping is wanted, the last keyframe should
-                // be the same as the first one
-                Vec3::new(1.0, 0.0, 1.0),
-            ],
-        ),
+        UnevenSampleAutoCurve::new([0.0, 1.0, 2.0, 3.0, 4.0].into_iter().zip([
+            Vec3::new(1.0, 0.0, 1.0),
+            Vec3::new(-1.0, 0.0, 1.0),
+            Vec3::new(-1.0, 0.0, -1.0),
+            Vec3::new(1.0, 0.0, -1.0),
+            // in case seamless looping is wanted, the last keyframe should
+            // be the same as the first one
+            Vec3::new(1.0, 0.0, 1.0),
+        ]))
+        .map(TranslationCurve)
+        .expect("Failed to build translation curve"),
```

Comment:
nit: since this an example I'll be strict on following rust api guidelines, to encourage good practices
```suggestion
        .expect("failed to build translation curve"),
```